server/src/main/java/org/opensearch/search/query/TopDocsCollectorContext.java

```diff
@@ -353,7 +365,15 @@ protected ReduceableSearchResult reduceWith(final Collection<CollapseTopFieldDoc
                     numHits,
                     topFieldDocs.toArray(new CollapseTopFieldDocs[0])
                 );
-                result.topDocs(new TopDocsAndMaxScore(topDocs, maxScore), sortFmt);
+                TopDocsAndMaxScore topDocsAndMaxScore;
```

Comment:
I don't think we need this logic here, the `maxScore` is already pre-calculated, right? We should just use it I think

Reply:
This change is for the case when concurrent segment search enabled, since we omit the maxScoreCollector in line 320, max score is Float.NaN now, so we need to get the max score from the first matched document here. Line 281 to line 290 is for the case that concurrent segment search is not enabled, this change within the createManager() is for concurrent segment search is enabled.

Reply:
> The reduceWith() method is only called when concurrent search enabled,

You are absolutely right, I just realized that we should not see it called in non-concurrent search